File: sleeping_in_class.py
# import sys
# sys.stdin
# sys.stdout
"""
PYTHON SOLUTION**

**Solution not tested

USACO 2022 February Contest, Bronze
Problem 1. Sleeping in Class

link the the problem: 
http://usaco.org/index.php?page=viewproblem2&cpid=1203
"""

num_cases = int(input())
case_lens = []
cases = []

for number in range(num_cases):
    case_lens.append(int(input()))
    cases.append([int(number) for number in input().split()])

modifications = []

for case in cases:
    divs = []
    cur_range = []
    sum_case = sum(case)
    #create list of factors
    for number in range(max(case), sum_case + 1):
        if sum_case == 0:
            modifications.append(0)

        elif sum_case % number == 0:
            target_factor = number
            cur_sum = 0
            possible = True
            for nap in case:
                cur_sum += nap
                if cur_sum > target_factor:
                    possible = False
                    break
                elif cur_sum == target_factor:
                    cur_sum = 0
            
            if possible == True:
                modifications.append(len(case) - sum_case/target_factor)
                break

for item in modifications:
    print(int(item))

File: bronze/sleepinginclass.cpp
/*
C++ SOLUTION

USACO 2022 February Contest, Bronze
Problem 1. Sleeping in Class
Link to problem:
http://usaco.org/index.php?page=viewproblem2&cpid=1203 
*/

#include <iostream>
#include <string>
#include <vector>
#include <algorithm>
using std::cin;
using std::cout;
using std::endl;
using std::vector;
using std::sort;



int T;

void solve() {
    int N, sum = 0; 
    cin >> N;
    vector<int> nums(N);
    for (int& num:nums) {
        cin >> num;
        sum += num;
    }
    for (int num = N; num > 0; num--) {
        if (sum % num == 0) {
            bool ok = true;
            int target = sum / num;
            int current = 0;
            for (int num:nums) {
                current += num;
                if (current > target) {
                    ok = false;
                    break;
                }
                if (current == target) {
                current = 0;
                }
            } 
            if (ok) {
                cout << N - num << endl;
                break;
            }
    }}
}

int main() {
    cin >> T;
    while (T--) {
        solve();
    }
}